warehouse/core/src/main/java/datawave/iterators/filter/ageoff/DataTypeParser.java

```diff
@@ -0,0 +1,154 @@
+package datawave.iterators.filter.ageoff;
+
+import org.apache.accumulo.core.data.ArrayByteSequence;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * Originally extracted from DataTypeAgeOffFilter
+ *
+ * Contains a single public static method for parsing the datatype (as a ByteSequence) from a key.
+ *
+ * It can parse the various shard schema key structures including the d, fi, tf, and event schemas.
+ *
+ * It can also parse an index table key.
+ */
+public class DataTypeParser {
+    
+    private static final char FI_CF_CHAR_1 = 'f';
+    private static final char FI_CF_CHAR_2 = 'i';
+    
+    private static final char TF_CF_CHAR_2 = 'f';
+    private static final char TF_CF_CHAR_1 = 't';
+    
+    private static final char DOC_CF_CHAR = 'd';
+    
+    private static final int NULL_BYTE = 0x00;
+    
+    // shard ID's minimum length = sum of the minimum length of its components: year + month + day + underscore + single digit partition
+    private static final int MIN_LENGTH_SHARD_ID = 10;
+    
+    /**
+     * @param key
+     *            Accumulo key to parse
+     * @param isIndexTable
+     *            true if the key is from an index table, false if the shard table
+     * @return a ByteSequence containing the data type name, possibly returning null or empty for unexpected column data.
+     */
+    public static final ByteSequence parseKey(Key key, boolean isIndexTable) {
+        /**
+         * Supports the shard and index table. There should not be a failure, however if either one is used on the incorrect table
+         */
+        if (isIndexTable) {
+            return extractDatatypeForIndexTable(key);
+        } else {
+            return extractDatatypeForShardTable(key);
+        }
+    }
+    
+    private static final ByteSequence extractDatatypeForShardTable(Key k) {
+        // ASSUMES THAT THE KEY STARTS WITH A CORRECTLY SIZED BYTE ARRAY
+        byte[] cf = k.getColumnFamilyData().getBackingArray();
```

Comment:
for my own understanding, is there any worry about a situation where the byte array is of an incorrect size?

Reply:
I left this comment intact from the legacy code that was moved here.  To me, it seems like this is a reasonable assumption.  If the key doesn't start with a correctly sized byte array, then I think it's reasonable if an exception were thrown.